plugins/file/index.js

```diff
@@ -13,15 +13,16 @@ function getDateString() {
   return `${now.getFullYear()}/${month}/${day}`;
 }
 
-async function upload(s3, params) {
-  return new Promise((resolve, reject) => {
-    s3.upload(params, (err, data) => {
-      if (err) {
-        reject(err);
-      }
-      resolve(data);
-    });
-  });
+async function upload(s3Client, params, s3Region) {
+  try {
+    await s3Client.send(new PutObjectCommand(params));
+    return {
+      key: params.Key,
+      url: `https://${params.Bucket}.s3.${s3Region}.amazonaws.com/${params.Key}`,
+    };
+  } catch (error) {
+    throw error;
+  }
```

Comment:
```suggestion
  try {
    const data = await s3Client.send(new PutObjectCommand(params));
    return {
      key: data.Key,
      url: `https://${params.Bucket}.s3.${s3Region}.amazonaws.com/${data.Key}`,
    };
  } catch (error) {
    throw error;
  }
```

Reply:
There's no "Key" property in data. Had to revert that change again.

Reply:
Ah okey, I understand 👍